### DISCUSSION TITLE
Questions about the RescaleOutput

### DISCUSSION THREAD

**Discussion by yugang-hello** [mir-group/nequip#333] [Q&A] [answered]
Dear developers:
    Thanks for your amazing code!
    As the default options of nequip, the per_species_rescale will be set to dataset_per_atom_total_energy_mean (for shifts) and dataset_forces_rms (for scales). While the global_rescale will also be used with null (for shifts), and dataset_forces_rms (for scales). Several question to that:
    1. Why the global_rescale (RescaleOutput) should be used, and only used for model.eval()? It seems like the per_species_rescale (PerSpeciesScaleShift) is enough.
    2. It seems like when the RescaleOutput is used, the update_for_rescale of PerSpeciesScaleShift will be called, and the dataset_forces_rms of per_species_rescale seems to be invalided. At the same time, the RescaleOutput won't be used during training. I just wonder about this option.
    Thank you for your time and expertise, and I look forward to hearing back from you :)

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @yugang-hello ,

Thanks for your interest in our work!

1. We do this in order to do training in normalized units, but be able to do inference or validation in our real physical units.
2. This is because PerSpeciesScaleShift operates in "normalized" units rather than physical units, but if its scales/shifts are provided in physical units, they have to be scaled into normalized units. This is what `update_for_rescale` does

**Comment by yugang-hello** [comment]
Dear Linux-cpp-lisp:
Really thanks for your explanation.
Now I can understand, it is a useful design 👍.

Best wishes!
